### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ik-web**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const tempObject = Object.assign({}, state);
```
Comment (illiarerun) [inline src/transformStateWithClones.js:11]: you can use spread operator here

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
```
Comment (illiarerun) [inline src/transformStateWithClones.js:10]: please, don't use names like result/array etc., make names meaningful

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,35 @@
...
+  const tempObject = Object.assign({}, state);
+
+  for (const obj of actions) {
+    switch (obj.type) {
```
Comment (illiarerun) [inline src/transformStateWithClones.js:14]: [Checklist ](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) 3rd item, use object destructuring

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,35 @@
...
+        }
+        break;
+
+      case 'clear':
+        for (const key in tempObject) {
+          delete tempObject[key];
+        }
+    }
+
+    result.push(Object.assign({}, tempObject));
```
Comment (illiarerun) [inline src/transformStateWithClones.js:35]: same here, you can use spread operator

File: src/transformStateWithClones.js [hunk 5]
```diff
@@ -7,7 +7,35 @@
...
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(tempObject, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const el of obj.keysToRemove) {
+          for (const key in tempObject) {
```
Comment (illiarerun) [inline src/transformStateWithClones.js:21]: why do you need this loop? you can just check if property is in object with .hasOwnProperty method

**Review by illiarerun — CHANGES_REQUESTED** [review]
GJ, now let's work on fixes:

File: src/transformStateWithClones.js (hunk 1 above)
Comment (ik-web) [reply src/transformStateWithClones.js]: Fixed

File: src/transformStateWithClones.js (hunk 2 above)
Comment (ik-web) [reply src/transformStateWithClones.js]: Fixed

File: src/transformStateWithClones.js (hunk 3 above)
Comment (ik-web) [reply src/transformStateWithClones.js]: Fixed

File: src/transformStateWithClones.js (hunk 4 above)
Comment (ik-web) [reply src/transformStateWithClones.js]: Fixed

File: src/transformStateWithClones.js (hunk 5 above)
Comment (ik-web) [reply src/transformStateWithClones.js]: Fixed

**Review by jstmpelowycz — APPROVED** [review]
nice 🔥

**Pull request closed without merging**